### DISCUSSION TITLE
Does webrtc supports dynamic bitrate ?

### DISCUSSION THREAD

**Discussion by varmavishal** [paullouisageneau/libdatachannel#624] [Q&A]
Hello,
I'm using your code from libdatachannel/examples/media-sender/ and have a few questions.
1. Does webrtc adjust the bitrate dynamically based on the minimum and maximum rates? 
2. How does the encoder receive the information? Is there a channel from the decoder to the encoder that reports the number of packets received?

Regards.

**Comment by paullouisageneau** [comment]
>     1. Does webrtc adjust the bitrate dynamically based on the minimum and maximum rates?

WebRTC implementations can support dynamically adjusting the bitrate with extensions like REMB and TWCC. However, the encoding pipeline must support it. The media-sender example is very basic and adjusting the bitrate dynamically is not supported, it set once and for all when starting gstreamer.

>     2. How does the encoder receive the information? Is there a channel from the decoder to the encoder that reports the number of packets received?

The encoder gets the information with RTCP feedback, i.e. RTCP packets sent back to the sender by the receiver. In the media-sender example, incoming RTCP packets are ignored.
